led_strip/include/led_strip_spi.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2022-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdint.h>
+#include "esp_err.h"
+#include "hal/spi_types.h"
```

Comment:
Better to rely on the driver, not the hal.

```suggestion
#include "driver/spi_master.h"
```